apps/juicebox_web/assets/js/reactions/recorder.js

```diff
@@ -0,0 +1,85 @@
+/* global [MediaRecorder] */
+
+import React, { Component } from 'react';
+import { delay } from 'lodash';
+import VideoEncoder from './video-encoder';
+
+class ReactionRecorder extends Component {
+  render() {
+    return (
+      <div>
+        <video ref={(video) => this.video = video}></video>
+        <button onClick={_record.bind(this)}>Reaction!</button>
+      </div>
+    );
+  }
+}
+
+function _startRecording(stream) {
+  const send = this.props.onRecord.bind(this);
+
+  const encoder = new VideoEncoder();
+  let mediaRecorder;
+
+  if (typeof MediaRecorder.isTypeSupported == 'function'){
+    let options;
+
+    if (MediaRecorder.isTypeSupported('video/webm;codecs=h264')) {
+      options = {mimeType: 'video/webm;codecs=h264'};
+    } else if (MediaRecorder.isTypeSupported('video/webm;codecs=vp9')) {
+      options = {mimeType: 'video/webm;codecs=vp9'};
+    } else if (MediaRecorder.isTypeSupported('video/webm;codecs=vp8')) {
+      options = {mimeType: 'video/webm;codecs=vp8'};
+    }
+    mediaRecorder = new MediaRecorder(stream, options);
+  }else{
+    mediaRecorder = new MediaRecorder(stream);
```

Comment:
Possible undefined variable: 'MediaRecorder' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)